docs/azure/sdk/authentication/create-token-credentials-from-configuration.md

```diff
@@ -0,0 +1,151 @@
+* [DefaultAzureCredential](#create-a-defaultazurecredential-type)
+
+Add the wildcard value "*" to allow the credential to acquire tokens for any Microsoft Entra tenant the logged in account can access. If no tenant IDs are specified, this option will have no effect on that authentication method, and the credential will acquire tokens for any requested tenant when using that method.
+
+```json
+{
+    "additionallyAllowedTenants":  "<tenant-ids-separated-by-semicolon>"
+}
+```
+
+### Create a `ManagedIdentityCredential` type
+
+You can create both user-assigned and system-assigned managed identities using configuration values. Add the following configuration values to your _appsettings.json_ file to create an instance of <xref:Azure.Identity.ManagedIdentityCredential?displayProperty=fullName>.
+
+#### User-assigned managed identities
+
+Specify a user-assigned managed identity via a client ID:
+
+```json
+{
+    "credential": "managedidentity",
+    "clientId":  "<clientId>"
+}
+```
+
+Specify a user-assigned managed identity via a resource ID:
+
+```json
+{
+    "credential": "managedidentity",
+    "managedIdentityResourceId":  "<managedIdentityResourceId>"
+}
+```
+
+The resource ID takes the form `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}`.
+
+#### System-assigned managed identities
+
+```json
+{
+    "credential": "managedidentity"
+}
+```
+
+### Create a `WorkloadIdentityCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.WorkloadIdentityCredential?displayProperty=fullName>:
+
+```json
+{
+    "credential": "workloadidentity",
+    "tenantId":  "<tenantId>",
+    "clientId":  "<clientId>",
+    "tokenFilePath": "<tokenFilePath>"
+}
+```
+
+### Create a `ClientSecretCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ClientSecretCredential?displayProperty=fullName>:
+
+```json
+{
+    "tenantId":  "<tenantId>",
+    "clientId":  "<clientId>",
+    "clientSecret": "<clientSecret>"
+}
+```
+
+### Create a `ClientCertificateCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ClientCertificateCredential?displayProperty=fullName>:
+
+```json
+{
+    "tenantId":  "<tenantId>",
+    "clientId":  "<clientId>",
+    "clientCertificate": "<clientCertificate>",
+    "clientCertificateStoreLocation": "<clientCertificateStoreLocation>",
+    "additionallyAllowedTenants": "<tenant-ids-separated-by-semicolon>"
+}
+```
+
+> [!NOTE]
+> The `clientCertificateStoreLocation` and `additionallyAllowedTenants` values are optional. If no `clientCertificateStoreLocation` is specified, the default `CurrentUser` is used from the <xref:System.Fabric.X509Credentials.StoreLocation?displayProperty=nameWithType> enum.
```

Comment:
What happens if the `clientCertificateStoreLocation` and `additionallyAllowedTenants` keys are present, but their values are empty strings? Is that treated the same as the absence of both key-value pairs? It would be good to say here.